### PR TITLE
Use render_component in render_inline test helper

### PR DESCRIPTION
**Pull request by czj**
### Summary

Using `render_component` in the the `render_inline` test helper instead of `render` allows tests to pass on any version of Rails (>= 6.1 and < 6.1 with or without monkey patch always pass.

Without this fix, when running Rails < 6.1 without the monkey patch, using `render_inline`would result in tests failure.

Closes https://github.com/github/view_component/issues/396

### CODE REVIEW

File: lib/view_component/test_helpers.rb [hunk 1]
```diff
@@ -20,7 +20,7 @@ def refute_component_rendered
     attr_reader :rendered_component
 
     def render_inline(component, **args, &block)
-      @rendered_component = controller.view_context.render(component, args, &block)
+      @rendered_component = controller.view_context.render_component(component, &block)
```
Comment (joelhawksley) [inline lib/view_component/test_helpers.rb:23]:
```suggestion
      @rendered_component = 
        if if Rails.version.to_f >= 6.1
          controller.view_context.render(component, &block)
        else
          controller.view_context.render_component(component, &block)
        end
```

**Comment by joelhawksley** [comment]
@czj one suggested change so that this will work on 6.1, where `render_component` isn't available. 

Can you add a CHANGELOG entry and your name to the README?

File: lib/view_component/test_helpers.rb (hunk 1 above)
Comment (krzysiek1507) [reply lib/view_component/test_helpers.rb:23]: Wouldn't it be better to check `Rails::VERSION::MAJOR >= 6 && Rails::VERSION::MINOR >= 1`?

**Comment by czj** [comment]
Thanks @joelhawksley, changes have been applied.

I have one last question : is there a reason why the `render_component` method is not made available for Rails >= 6.1 ? I searched into commits and issues but did not find one.

For me, this method would be useful in any version of Rails, for two reasons :

- you can search your code for any ViewComponent render by using /render_component/ instead of doing a less reliable search for /Component\./
- using `render_component` should be faster and more memory efficient : you skip all the complexity of Rails' "all-in-one" `render` method (pass less arguments, instantiate less objects in memory, go through less method calls)

That seems like a sane alternative to me.

**Comment by joelhawksley** [comment]
@czj https://github.com/github/view_component/pull/358#issuecomment-652882373 mentions the decision, but doesn't give a reason.

@johannesengl and I debated it a bit while pairing, and came back to the [design philosophy](https://github.com/github/view_component#design-philosophy) for the library, which is to integrate with Rails as cleanly as possible. I don't feel that adding an explicit `render_component` method is in line with that philosophy.

That being said, I'd be open to adding a configuration option to enable `render_component` in Rails 6.1+. Would you like to implement that in another PR?

File: README.md [hunk 2]
```diff
@@ -929,10 +929,10 @@ ViewComponent is built by:
 |@maxbeizer|@franco|@tbroad-ramsey|@jensljungblad|@bbugh|
 |Nashville, TN|Switzerland|Spring Hill, TN|New York, NY|Austin, TX|
 
-|<img src="https://avatars.githubusercontent.com/johannesengl?s=256" alt="johannesengl" width="128" />|
+|<img src="https://avatars.githubusercontent.com/johannesengl?s=256" alt="johannesengl" width="128" />|<img src="https://avatars.githubusercontent.com/czj?s=256" alt="czj" width="128" />|
 |:---:|
```
Comment (joelhawksley) [inline README.md:933]:
```suggestion
|:---:|:---:|
```

**Pull request merged**